04 - Guides, Workflows, & Courses/Guides/Want some Sass with your obsidian theme‽ here's How and Why.md

```diff
@@ -0,0 +1,221 @@
+---
+aliases:
+- sass
+- scss
+tags:
+- 
+publish: true
+---
+# Want some Sass with your obsidian theme? Here's How and Why
+*written by [[jdanielmourao]]*
+
+## What is Sass and Scss?
+Sass, aka **S**yntactically **A**wesome **S**tyle**S**heet, is a CSS pre-processor. It has features such as nesting, mixins, and more, which allow you to make cleaner CSS code and reduce repetitions, resulting in smaller files, and making the CSS file easier to maintain. Do keep in mind, though, that using a pre-processor is only advisable **after** you've learned the basics of CSS, and are familiar with general good practices for CSS.
+
+What is a pre-processor exactly? Well, to put it simply, it's a program that processes an input file to produce an output that is used for other programs. In this case, it takes a `.sass`/`.scss` file, and outputs it as `.css`.
+
+### What's the difference between the two?
+Both Sass and Scss are two different syntaxes that achieve the same thing. One is an extension of the CSS syntax (SCSS), while the other, older syntax, is a simpler way of writing (Indented Syntax, commonly known as Sass). Note that the old indented syntax is often referred to as Sass because it was the first to exist, but nowadays, when referring to Sass, people mean the pre-processor (and even Scss), and not the indented syntax.
+
+Here's a simple example:
+
+SCSS:
+```scss
+.nav-action-button {
+    color: grey;
+    height: 4px;
+    
+    &.is-active {
+        background-color: black;
+        color: white;
+    }
+    &:hover {
+        color: unset;
+        background-color: blue;
+    }
+}
+```
+Indented syntax (Sass):
+```scss
+.nav-action-button
```

Comment:
If it's the codeblocks, I had it with scss for both because obsidian didn't let me put sass as a language, and I still wanted the syntax highlighting. Both could just be regular code blocks, though. The main difference is that one requires `;` and `{`, while the other barely uses any punctuation.

Reply:
OK thanks - it is easier to see on a larger screen.